src/openai/lib/azure.py

```diff
@@ -287,8 +287,7 @@ def _prepare_options(self, options: FinalRequestOptions) -> None:
 
         azure_ad_token = self._get_azure_ad_token()
         if azure_ad_token is not None:
-            if headers.get("Authorization") is None:
-                headers["Authorization"] = f"Bearer {azure_ad_token}"
+            headers["Authorization"] = f"Bearer {azure_ad_token}"
```

Comment:
I'm slightly nervous that this means we'll override any `Authorization` header that a user set which wouldn't be what they wanted.

I think we could restructure how this `_prepare_options()` method is called slightly to avoid this instead, I'll look into it.

Reply:
Put up a PR here: https://github.com/openai/openai-python/pull/1533